File: Reservations-app/src/app/models/User.model.ts
import { Reservation } from "./Reservation.model";
import { Friend } from "./Friend.model";

export class UserModel {
  constructor(
    public firstName: string,
    public lastName: string,
    public email: string,
    public password: string,
    public image: string,
    public city: string,
    public street: string,
    public phoneNumber: string,
    public status: string, //1->head admin 2->car-admin 3->avio-admin 4->user
    public friends: Friend[] = [],
    public reservations: Reservation[] = [],
    public id?: number,
    public Id?: string,
    public passportNo?: string
  ) {}
}

File: ReservationAPI/ReservationAPI/Models/UserModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ReservationAPI.Models
{
    public class UserModel
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string Street { get; set; }
        public string City { get; set; }
        public string Phone { get; set; }
        public string Password { get; set; }

    }
}
